### PR TITLE
DOC: Improve the docstring of Str.contains()

### PR DESCRIPTION
**Pull request by Blair-Young**
- [ ] closes #xxxx
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

```############################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameter "flags" description should start with capital letter
```
`Achieved through PyData 2018 pandas sprint.`

### CODE REVIEW

**Comment by Blair-Young** [comment]
Unsure best way to phrase `flags` parameter as description starts with the ```re``` module.

File: pandas/core/strings.py [hunk 1]
```diff
@@ -295,20 +295,24 @@ def str_count(arr, pat, flags=0):
 
 def str_contains(arr, pat, case=True, flags=0, na=np.nan, regex=True):
     """
-    Return boolean Series/``array`` whether given pattern/regex is
-    contained in each string in the Series/Index.
+    Test if pattern or regex is contained within each string of a Series.
```
Comment (WillAyd) [inline pandas/core/strings.py]: This method is used by `Index` as well which was removed here - please be sure to add that back

File: pandas/core/strings.py [hunk 2]
```diff
@@ -295,20 +295,24 @@ def str_count(arr, pat, flags=0):
...
+
+    Return boolean Series based on whether a given pattern or regex is
```
Comment (WillAyd) [inline pandas/core/strings.py]: Related to comment above, this isn't always true

```python
index = pd.Index(['foo>>> index = pd.Index(['foo', 'bar', 'baz'])
>>> index.str.contains('a')
array([False,  True,  True])
```

File: pandas/core/strings.py [hunk 3]
```diff
@@ -295,20 +295,24 @@ def str_count(arr, pat, flags=0):
...
-    na : default NaN, fill value for missing values.
+        re module flags, e.g. re.IGNORECASE.
+    na : default NaN
+        Fill value for missing values.
     regex : bool, default True
-        If True use re.search, otherwise use Python in operator
+        If True, assumes the passed-in pattern is a regular expression.\n
+        If False, treats the pattern as a literal string.
 
     Returns
```
Comment (WillAyd) [inline pandas/core/strings.py]: Maybe use the Returns section from the below PR as reference

https://github.com/pandas-dev/pandas/pull/20491/files

File: pandas/core/strings.py [hunk 4]
```diff
@@ -295,20 +295,24 @@ def str_count(arr, pat, flags=0):
...
+        If True, case sensitive.
     flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
-    na : default NaN, fill value for missing values.
+        re module flags, e.g. re.IGNORECASE.
+    na : default NaN
+        Fill value for missing values.
     regex : bool, default True
-        If True use re.search, otherwise use Python in operator
+        If True, assumes the passed-in pattern is a regular expression.\n
```
Comment (WillAyd) [inline pandas/core/strings.py]: Rogue newline?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for submitting this

File: pandas/core/strings.py [hunk 5]
```diff
@@ -295,20 +295,24 @@ def str_count(arr, pat, flags=0):
...
-    case : boolean, default True
-        If True, case sensitive
+    pat : str
+        Character sequence or regular expression.
+    case : bool, default True
+        If True, case sensitive.
     flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
-    na : default NaN, fill value for missing values.
+        re module flags, e.g. re.IGNORECASE.
```
Comment (WillAyd) [inline pandas/core/strings.py]: To suppress the warning just rewrite this as "Flags to pass through to the re module, e.g. re.IGNORECASE." or something similar

File: pandas/core/strings.py [hunk 6]
```diff
@@ -318,6 +322,48 @@ def str_contains(arr, pat, case=True, flags=0, na=np.nan, regex=True):
...
+
+    >>> s.str.contains('og', regex=False)
+    0    False
+    1     True
+    2    False
+    3    False
+    4      NaN
+    dtype: object
+
+    >>> s.str.contains('og', na=False, regex=True)
```
Comment (WillAyd) [inline pandas/core/strings.py]: These examples are good but some text to direct the user on what to look at preceding each would be helpful. For example here I would prepend something like:

```With the above examples, you'll note that NA values in the caller return `np.nan` by default and as a result cast the dtype of the returned object to `object`. You can control this behavior by specifying a an alternate `na` argument```

File: pandas/core/strings.py [hunk 7]
```diff
@@ -318,6 +322,48 @@ def str_contains(arr, pat, case=True, flags=0, na=np.nan, regex=True):
     --------
     match : analogous, but stricter, relying on re.match instead of re.search
 
+    Examples
```
Comment (WillAyd) [inline pandas/core/strings.py]: Can you add examples for the `case` and `flags` parameters?

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by Blair-Young** [comment]
Hey @WillAyd I've tried to fix your correction. What's the next step?

**Review by jreback — APPROVED** [review]
lgtm. @TomAugspurger @jorisvandenbossche

**Review by WillAyd — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks @Blair-Young!

**Comment by Blair-Young** [comment]
Thanks @TomAugspurger looking forward to more.

**Pull request merged**